internal/configconverter/normalize_gcp.go

```diff
@@ -0,0 +1,78 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package configconverter
+
+import (
+	"context"
+	"fmt"
+	"go.opentelemetry.io/collector/confmap"
+	"log"
+)
+
+type NormalizeGcp struct{}
+
+func (NormalizeGcp) Convert(_ context.Context, in *confmap.Conf) error {
+	if in == nil {
+		return fmt.Errorf("cannot NormalizeGcp on nil *confmap.Conf")
+	}
+
+	const resourceDetector = "processors::resourcedetection::detectors"
```

Comment:
components can be named so you'll want to match against an optional `/<name>`. Otherwise we could just get the sub-config and not need to iterate.